### ISSUE TITLE
How to compute loss during evaluation?

### ISSUE THREAD

**Issue by Igal20** [facebookresearch/maskrcnn-benchmark#178]
## ❓ Questions and Help
How would you recommend  to compute loss during evaluation?
for example I set my model to evaluation --> model.eval()

# get prediction 
with torch.no_grad():
            predictions = model(images)
# calculate loss between **predictions** and **targets**
**SOMETHING LIKE THAT**
criterion = nn.crossentropyloss()
loss = criterion(predictions, targets)

**Comment by Igal20** [comment]
My goal is to see convergence of loss of train dataset and validation dataset.
Something like that.
Obviously I can not set model to `train` mode because than my model will learn validation set as well.

Is there other way to do it?

![diagnostic-line-plot-showing-a-good-fit-for-a-model](https://user-images.githubusercontent.com/20500897/48720963-93a9e780-ec29-11e8-8477-81b8d43eac36.png)

**Comment by fmassa** [comment]
You can plot the validation accuracy, this is probably going to be a good-enough metric for you, and that's what we usually do here.
So you'd plot training loss and validation accuracy.

How does that sound?

**Comment by Igal20** [comment]
Sounds good, how do one does it?

**Comment by fmassa** [comment]
You can evaluate your model at every checkpoint (with `tools/test_net.py`), and that will give your almost all you need.

**Comment by Igal20** [comment]
Only problem that I have is that `training loss` and `validation accuracy` are in different scales so I can not plot them on one scale.
`**training loss**`  ==> Sum of  `loss_classifier` + `loss_box_reg` + `loss_mask` + `loss_objectness` + `loss_rpn_box_reg`
**While**
`**validation accuracy**`  ==> mAP (mean Average Precision)

How do you handle with it?

**Comment by fmassa** [comment]
You can normalize it to be in the same range.
Also, the validation accuracy is expected to go up, while the training loss should go down.
